app/views/project_dataset_levels/_approval.html.erb

```diff
@@ -4,8 +4,8 @@
 <td id="approvals_<%=dom_id(project_dataset_level)%>">
 <% if ProjectDataset.dataset_approval(current_user).include? project_dataset_level.project_dataset %>
   <% if project_dataset_level.approved.nil? %>
-    <%= link_to bootstrap_icon_tag('ok'), approve_project_dataset_level_path(id: project_dataset_level.id, "project_dataset_level[approved]" => 'true'), :method => :patch, remote: true, id: "approval_#{dom_id(project_dataset_level)}", class: 'btn btn-success btn-sm  rce_item' %>
-    <%= link_to bootstrap_icon_tag('remove'), project_dataset_level_path(id: project_dataset_level.id, "project_dataset_level[approved]" => 'false'), :method => :patch, remote: true, class: 'btn btn-danger btn-sm reject_data_source_item' %>
+    <%= link_to bootstrap_icon_tag('ok'), approve_project_dataset_level_path(id: project_dataset_level.id, "project_dataset_level[decision_date]" => Time.zone.today, "project_dataset_level[approved]" => 'true'), :method => :patch, remote: true, id: "approval_#{dom_id(project_dataset_level)}", class: 'btn btn-success btn-sm  rce_item' %>
+    <%= link_to bootstrap_icon_tag('remove'), project_dataset_level_path(id: project_dataset_level.id, "project_dataset_level[decision_date]" => Time.zone.today, "project_dataset_level[approved]" => 'false'), :method => :patch, remote: true, class: 'btn btn-danger btn-sm reject_data_source_item' %>
```

Comment:
How much do/can you trust end users? 

Realistically, there's probably little to gain in an enterprising user manipulating the value that gets sent to the server in this instance (that I'm aware of), but something to bear in mind.